src/io/ipc/convert.rs

```diff
@@ -657,6 +658,9 @@ pub(crate) fn get_fb_field_type<'a>(
                 children: Some(fbb.create_vector(&children)),
             }
         }
+        Extension(ex) => {
+            todo!("extension");
```

Comment:
here is where we map the extension to the IPC field. To see how this is done by the rest of the arrow implementations, write 

```rust
    #[test]
    fn read_generated_extension() -> Result<()> {
        test_file("1.0.0-littleendian", "generated_extension")
    }
```
on `tests/it/io/ipc/file.rs` and print the schema in the json format (i..e. `arrow_json` inside the function `read_gzip_json`. You will find a small `ARROW:extension:...` key and value, which is how extensions are written in the metadata.